app/forms/funding/bursary_form.rb

```diff
@@ -3,65 +3,85 @@
 module Funding
   class BursaryForm < TraineeForm
     FIELDS = %i[
+      applying_for_scholarship
       applying_for_bursary
       bursary_tier
-      tiered_bursary_form
+      funding_type
     ].freeze
 
     NON_TRAINEE_FIELDS = %i[
-      tiered_bursary_form
+      funding_type
     ].freeze
 
-    attr_accessor(*FIELDS)
+    BURSARY_TYPE = "bursary"
+    SCHOLARSHIP_TYPE = "scholarship"
+    NONE_TYPE = "none"
+    FUNDING_TYPES = (Trainee.bursary_tiers.keys +
+                     [BURSARY_TYPE, SCHOLARSHIP_TYPE, NONE_TYPE]).freeze
 
-    before_validation :set_applying_for_bursary
+    attr_accessor(*FIELDS)
 
-    validates :applying_for_bursary, inclusion: { in: [true, false] }
-    validates :bursary_tier, inclusion: { in: Trainee.bursary_tiers.keys }, if: :requires_bursary_tier?
+    validates :funding_type, inclusion: { in: FUNDING_TYPES }
 
-    def applying_for_bursary=(value)
-      @applying_for_bursary = ActiveModel::Type::Boolean.new.cast(value)
-    end
+    delegate :can_apply_for_scholarship?, :can_apply_for_tiered_bursary?,
+             :scholarship_amount, to: :funding_manager
 
-    def save!
-      if valid?
-        update_trainee_attributes
-        trainee.save!
-        clear_stash
-      else
-        false
-      end
+    def initialize(trainee, params: {}, user: nil, store: FormStore)
+      params = add_fields_from_params(params)
+      super(trainee, params: params, user: user, store: store)
     end
 
   private
 
-    def update_trainee_attributes
-      # Need to save the applying_for_bursary attribute
-      trainee.assign_attributes(
-        fields
-          .merge(applying_for_bursary: applying_for_bursary)
-          .except(*fields_to_ignore_before_stash_or_save),
-      )
+    def compute_fields
+      opts = trainee.attributes.symbolize_keys.slice(*FIELDS)
+      opts = add_funding_type_from_db(opts)
+      opts.merge!(new_attributes)
+      opts
     end
 
-    def set_applying_for_bursary
-      self.applying_for_bursary = true if bursary_tier.present?
+    def funding_manager
+      @funding_manager ||= FundingManager.new(trainee)
     end
 
-    def compute_fields
-      trainee.attributes.symbolize_keys.slice(*FIELDS).merge(new_attributes)
+    def add_funding_type_from_db(opts)
+      opts[:funding_type] =
+        if trainee.bursary_tier.present?
+          trainee.bursary_tier
+        elsif trainee.applying_for_bursary
+          BURSARY_TYPE
+        elsif trainee.applying_for_scholarship
+          SCHOLARSHIP_TYPE
+        elsif (trainee.applying_for_bursary == false) || (trainee.applying_for_scholarship == false)
+          NONE_TYPE
+        end
+      opts
     end
 
-    def form_store_key
-      :bursary
-    end
-
-    def requires_bursary_tier?
-      bursary_tier.present? || tier_not_selected?
+    def add_fields_from_params(opts)
```

Comment:
This method and `add_funding_type_from_db` look like they're starting to get a little out of hand - especially as we're going to be adding a third type of funding (grants) too. Since there's a push to get this out, let's leave it for now and consider again when we add grants.